Remove some obsolete ia64-related hacks from the strace source
08 Feb 2005 agriffis@gentoo.org

--- strace-4.5.8/process.c
+++ strace-4.5.8/process.c
@@ -82,14 +82,10 @@
 
 #ifdef HAVE_LINUX_PTRACE_H
 #undef PTRACE_SYSCALL
-# ifdef HAVE_STRUCT_IA64_FPREG
-#  define ia64_fpreg XXX_ia64_fpreg
-# endif
 # ifdef HAVE_STRUCT_PT_ALL_USER_REGS
 #  define pt_all_user_regs XXX_pt_all_user_regs
 # endif
 #include <linux/ptrace.h>
-# undef ia64_fpreg
 # undef pt_all_user_regs
 #endif
 
--- strace-4.5.8/signal.c
+++ strace-4.5.8/signal.c
@@ -53,14 +53,10 @@
 #endif
 #elif defined(HAVE_LINUX_PTRACE_H)
 #undef PTRACE_SYSCALL
-# ifdef HAVE_STRUCT_IA64_FPREG
-#  define ia64_fpreg XXX_ia64_fpreg
-# endif
 # ifdef HAVE_STRUCT_PT_ALL_USER_REGS
 #  define pt_all_user_regs XXX_pt_all_user_regs
 # endif
 #include <linux/ptrace.h>
-# undef ia64_fpreg
 # undef pt_all_user_regs
 #endif
 
--- strace-4.5.8/syscall.c
+++ strace-4.5.8/syscall.c
@@ -63,14 +63,10 @@
 #endif
 #elif defined(HAVE_LINUX_PTRACE_H)
 #undef PTRACE_SYSCALL
-# ifdef HAVE_STRUCT_IA64_FPREG
-#  define ia64_fpreg XXX_ia64_fpreg
-# endif
 # ifdef HAVE_STRUCT_PT_ALL_USER_REGS
 #  define pt_all_user_regs XXX_pt_all_user_regs
 # endif
 #include <linux/ptrace.h>
-# undef ia64_fpreg
 # undef pt_all_user_regs
 #endif
 
--- strace-4.5.8/system.c
+++ strace-4.5.8/system.c
@@ -32,6 +32,23 @@
 
 #include "defs.h"
 
+#ifdef HAVE_SYS_REG_H
+# include <sys/reg.h>
+# ifndef PTRACE_PEEKUSR
+#  define PTRACE_PEEKUSR PTRACE_PEEKUSER
+# endif
+# ifndef PTRACE_POKEUSR
+#  define PTRACE_POKEUSR PTRACE_POKEUSER
+# endif
+#elif defined(HAVE_LINUX_PTRACE_H)
+# undef PTRACE_SYSCALL
+# ifdef HAVE_STRUCT_PT_ALL_USER_REGS
+#  define pt_all_user_regs XXX_pt_all_user_regs
+# endif
+# include <linux/ptrace.h>
+# undef pt_all_user_regs
+#endif
+
 #ifdef LINUX
 #define _LINUX_SOCKET_H
 #define _LINUX_FS_H
--- strace-4.5.8/util.c
+++ strace-4.5.8/util.c
@@ -63,14 +63,10 @@
 # define PTRACE_PEEKUSR PTRACE_PEEKUSER
 #elif defined(HAVE_LINUX_PTRACE_H)
 #undef PTRACE_SYSCALL
-# ifdef HAVE_STRUCT_IA64_FPREG
-#  define ia64_fpreg XXX_ia64_fpreg
-# endif
 # ifdef HAVE_STRUCT_PT_ALL_USER_REGS
 #  define pt_all_user_regs XXX_pt_all_user_regs
 # endif
 #include <linux/ptrace.h>
-# undef ia64_fpreg
 # undef pt_all_user_regs
 #endif
 
